haystack/nodes/retriever/_openai_encoder.py

```diff
@@ -0,0 +1,144 @@
+import json
+import logging
+import sys
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Union
+
+import numpy as np
+import requests
+from tqdm.auto import tqdm
+
+from haystack.errors import OpenAIError, OpenAIRateLimitError
+from haystack.nodes.retriever._base_embedding_encoder import _BaseEmbeddingEncoder
+from haystack.schema import Document
+from haystack.utils.reflection import retry_with_exponential_backoff
+from haystack.utils.import_utils import _optional_component_not_installed
+
+if TYPE_CHECKING:
+    from haystack.nodes.retriever import EmbeddingRetriever
+
+USE_TIKTOKEN = False
+if sys.version_info >= (3, 8):
+    USE_TIKTOKEN = True
+
+if USE_TIKTOKEN:
+    try:
+        import tiktoken
+    except (ImportError, ModuleNotFoundError) as ie:
+        _optional_component_not_installed(__name__, "openai", ie)
```

Comment:
I don't think the error message printed by `_optional_component_not_installed` is relevant anymore since we added `tiktoken` as a core dependency. In fact I think we can completely remove the `try` and `except` clause since we should no longer get an `ImportError` since `USE_TIKTOKEN` is only true if we have a compatible python version.

Reply:
I removed the try-except block, as it's a core dependency now. But pylint didn't like it, so I had to disable the import error.